src/main/java/athena/TimeAllocator.java

```diff
@@ -23,35 +25,151 @@
-        this.fixedTask = tasks.getFilteredList(new FlexibleTimeFilter(false)).getTasks();
+        this.flexibleTaskList = tasks.getFilteredList(new FlexibleTimeFilter(true));
+        this.fixedTaskList = tasks.getFilteredList(new FlexibleTimeFilter(false));
+        this.taskList = tasks;
+
+    }
+
+    public void runAllocate() throws TaskNotFoundException {
+        LocalDate currDay = LocalDate.now();
+        ForecastFilter forecast = new ForecastFilter(Forecast.DAY);
+        for (int day = 0; day < 31; day++) {
+            System.out.println(day);
+            ArrayList<Integer> dayLog = new ArrayList<Integer>(Collections.nCopies(24, -1));
+            ArrayList<Task> predefinedTimeTasks = getSortedFixedTasks(getFixedDayTasks(currDay));
+            ArrayList<Task> undefinedTimeTasks = getSortedFixedTasks(getFlexibleDayTasks(currDay));
+
+            for (Task currTask : predefinedTimeTasks) {
+                int tag = currTask.getNumber();
+                Time timeInfo = currTask.getTimeInfo();
+                for (int i = 0; i < timeInfo.getDuration(); i++) {
+                    dayLog.set(timeInfo.getStartTime().getHour(), tag);
+                }
+            }
+
+            int start = 0;
+            int sleep = 24;
+            boolean done = false;
+            while (!done) {
+                int pos = start;
+                while (dayLog.get(pos) != -1) {
+                    pos++;
+                    if (pos == sleep) {
+                        done = true;
+                        break;
+                    }
+                }
+                int end = pos;
+                while (dayLog.get(end) == -1) {
+                    end++;
+                    if (end == sleep) {
+                        done = true;
+                        break;
+                    }
+                }
+                int space = end - pos;
+                ArrayList<Integer> temp = new ArrayList<Integer>(Collections.nCopies(space, -1));
+                ArrayList<Task> tasksToRemove = new ArrayList<Task>();
+                while (temp.contains(-1)) {
+                    for (Task currTask : undefinedTimeTasks) {
+                        Integer span = currTask.getTimeInfo().getDuration();
+                        if (span <= space) {
+                            int taskNumber = currTask.getNumber();
+                            this.flexibleTaskList.getTaskFromNumber(taskNumber)
+                                    .getTimeInfo().setStartTime(LocalTime.of(end - space, 0));
+                            for (int i = 0; i < span; i++) {
+                                temp.set(i + end - space - pos, currTask.getNumber());
+                            }
+                            space -= span;
+                            tasksToRemove.add(currTask);
+                            if (space == 0) {
+                                break;
+                            }
+                        }
+                    }
```

Comment:
Would be good to refactor this part into methods to maintain single level of abstraction.